src/main/java/org/cryptomator/windows/common/WindowsRegistry.java

```diff
@@ -0,0 +1,158 @@
+						pointerToResultKey,
+						NULL,
+						transactionHandle,
+						NULL
+				);
+				if (result != ERROR_SUCCESS()) {
+					throw new RegistryKeyException("winreg.h:RegCreateKeyTransactedW", key.getPath() + "\\" + subkey, result);
+				}
+				return new RegistryKey(pointerToResultKey.get(C_POINTER, 0), key.getPath() + "\\" + subkey);
+			}
+		}
+
+		public RegistryKey openRegKey(RegistryKey key, String subkey) throws RegistryKeyException {
+			var pointerToResultKey = Arena.ofAuto().allocate(AddressLayout.ADDRESS);
+			try (var arena = Arena.ofConfined()) {
+				var lpSubkey = arena.allocateFrom(subkey, StandardCharsets.UTF_16LE);
+				int result = Winreg_h.RegOpenKeyTransactedW(
+						key.getHandle(),
+						lpSubkey,
+						0,
+						KEY_READ() | KEY_WRITE(),
+						pointerToResultKey,
+						transactionHandle,
+						NULL
+				);
+				if (result != ERROR_SUCCESS()) {
+					throw new RegistryKeyException("winreg.h:RegOpenKeyTransactedW", key.getPath() + "\\" + subkey, result);
+				}
+				return new RegistryKey(pointerToResultKey.get(C_POINTER, 0), key.getPath() + "\\" + subkey);
+			}
+		}
+
+		public void deleteRegKey(RegistryKey key, String subkey) throws WindowsException {
+			deleteRegKey(key, subkey, false);
+		}
+
+		public void deleteRegKey(RegistryKey key, String subkey, boolean ignoreNotExisting) throws RegistryKeyException {
+			try (var arena = Arena.ofConfined()) {
+				var lpSubkey = arena.allocateFrom(subkey, StandardCharsets.UTF_16LE);
+				int result = Winreg_h.RegDeleteKeyTransactedW(
+						key.getHandle(),
+						lpSubkey,
+						0x0100, //KEY_WOW64_64KEY
+						0,
+						transactionHandle,
+						NULL
+				);
+				if (result != ERROR_SUCCESS() //
+						&& !(result == ERROR_FILE_NOT_FOUND() && ignoreNotExisting)) {
+					throw new RegistryKeyException("winreg.h:RegDeleteKeyTransactedW", key.getPath() + "\\" + subkey, result);
+				}
+			}
+		}
+
+
+		public synchronized void commit() throws WindowsException {
+			if (isClosed) {
+				throw new IllegalStateException("Transaction already closed");
+			}
+			int result = Ktmw32_h.CommitTransaction(transactionHandle);
+			if (result == 0) {
+				int error = Windows_h.GetLastError();
+				throw new WindowsException("ktmw32.h:CommitTransaction", error);
+			}
+			isCommited = true;
+			closeInternal();
+		}
+
+		public synchronized void rollback() throws WindowsException {
+			if (isClosed) {
+				throw new IllegalStateException("Transaction already closed");
+			}
+			int result = Ktmw32_h.RollbackTransaction(transactionHandle);
+			if (result == 0) {
+				int error = Windows_h.GetLastError();
+				throw new WindowsException("ktmw32.h:CommitTransaction", error);
+			}
+			closeInternal();
+		}
+
+		;
+
+		@Override
+		public synchronized void close() throws WindowsException {
+			if (!isCommited) {
+				try {
+					rollback();
+				} catch (WindowsException e) {
+					System.err.printf("Failed to rollback uncommited transaction on close: %s%n", e.getMessage());
```

Comment:
Use a logger

Reply:
Done in 541361925b65e150b2cdcf855cef6d9ba99a83d2